Parser/pegen_errors.c

```diff
@@ -250,8 +250,14 @@ get_error_line_from_tokenizer_buffers(Parser *p, Py_ssize_t lineno)
     char *cur_line = p->tok->fp_interactive ? p->tok->interactive_src_start : p->tok->str;
     assert(cur_line != NULL);
 
-    for (int i = 0; i < lineno - 1; i++) {
-        cur_line = strchr(cur_line, '\n') + 1;
+    Py_ssize_t relative_lineno = p->starting_lineno ? lineno - p->starting_lineno + 1 : lineno;
+
+    for (int i = 0; i < relative_lineno - 1; i++) {
+        char *new_line = strchr(cur_line, '\n') + 1;
+        if (new_line == NULL || new_line >= p->tok->inp) {
+            break;
```

Comment:
@pablogsal do we still need this condition or can we convert it to an `assert` (as a sanity check on the tokenizer's line numbers)? It seems like `relative_lineno` fixes this bug as is (I might be missing some context though).

Reply:
It depends on how you look at it. The reason I added the condition is that an error is reported as an incorrect line number and not a segfault, which looks much worse on release builds where there are no asserts.

Reply:
@isidentical If you want, I can put an assert here?

Reply:
> It depends on how you look at it. The reason I added the condition is that if we failed to do this correctly an error is reported as an incorrect line number and not a segfault, which looks much worse on release builds where there are no asserts.

Makes sense. It might be a bit repetitive but can we make this an assert as well? Since otherwise, it might hide bugs in the tokenizer and silently show the wrong line. If these sort of fuzzers use Python in debug mode (I don't know, but perhaps @gpshead knows) it would be an additional plus since they can catch it but the users won't get affected.

Reply:
> @isidentical If you want, I can put an assert here?

Ups, did not see this. Yep, I think it makes sense.

Reply:
I don't believe oss-fuzz has asserts enabled in cpython3 today based on https://github.com/google/oss-fuzz/tree/master/projects/cpython3 but we could append to contain `-UNDEBUG`. I've proposed that in https://github.com/google/oss-fuzz/pull/7119.